packages/react-components/src/components/table-output-component.tsx

```diff
@@ -603,6 +614,8 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
             });
             this.gridApi.redrawRows();
         }
+        this.gridSearched = true;
+        this.gridRedrawn = false;
```

Comment:
Good thinking, but doesn't seem to be enough. The problem is that `searchEvents` is triggered by `gridApi` every time a character is typed, but it changes the nodes on each pass, which delays the following triggers and changes. This makes it unreliable to control the `gridRedrawn` flag.
Maybe if we stop changing the nodes with `rowNode.data['isMatched'] = isMatched;`, and control the matched rows with a new independent array, then searchEvents would become fast enough to allow using `gridRedrawn` here.

Reply:
I think we should keep this promising yet major refactoring for another iteration or PR. It may unveil more issues while fixing the design at a broader scope.